metals/src/main/scala/scala/meta/internal/metals/testProvider/TestSuitesIndex.scala

```diff
@@ -0,0 +1,131 @@
+package scala.meta.internal.metals.testProvider
+
+import scala.collection.concurrent.TrieMap
+import scala.collection.mutable
+
+import scala.meta.internal.metals.debug.BuildTargetClasses
+import scala.meta.internal.metals.testProvider.TestExplorerEvent._
+import scala.meta.internal.mtags
+import scala.meta.internal.semanticdb.TextDocument
+import scala.meta.io.AbsolutePath
+
+import ch.epfl.scala.bsp4j.BuildTarget
+
+private[testProvider] final case class SymbolsPerTarget(
+    target: BuildTarget,
+    testSymbols: TrieMap[
+      BuildTargetClasses.Symbol,
+      BuildTargetClasses.FullyQualifiedClassName
+    ]
+)
+
+private[testProvider] final case class TestFileMetadata(
+    entries: List[TestEntry],
+    textDocument: Option[TextDocument]
```

Comment:
We can't keep all the text document in memory, that's why in References we use BloomFilter

Reply:
Good catch, I removed it and replaced it with `hasResolvedChildren` to keep info if this file is "relevant"